### PR TITLE
DROOLS-5683: Collection editor not opening for DRL based scenarios

### PR DESCRIPTION
**Pull request by yesamer**
**JIRA**: https://issues.redhat.com/browse/DROOLS-5683

**referenced Pull Requests**:

* https://github.com/kiegroup/drools/pull/3182

With this PR, now FrontEnd supports following new Java Classes
- `BigDecimal`
- `BigInteger`
- `Byte`
- `Character`
- `Float`
- `LocalDateTime`
- `LocalTime`
- `Long`
- `Short`

This means, in case of a `List<T>` where  `T` is one of the listed classes, Collection editor will throw an error with the current master version. The changes introduced with this PR will enable usage of these Java Object for CollectionEditor and as Simple Type.

Example with `List<LocalDateTime>`
Screenshot from 2020-10-22 15-25-40

Example with `List<BigDecimal>`  (comma is wrong as decimal separator, use period instead)
Screenshot from 2020-10-22 15-26-35

Example as Simple `LocalDateTime` object
Screenshot from 2020-10-22 15-24-06

Example as Simple `BigDecimal`object
Screenshot from 2020-10-22 15-22-21

`BigDecimal` & `BigInteger` support number grouping (eg. user can write `1,000,000.34` and it will be parsed as `1000000.34`)
`LocalDateTime` uses [ISO_LOCAL_DATE_TIME](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_LOCAL_DATE_TIME) which is `2011-12-03T10:15:30` format. It supports nanoseconds as well, so I added supports for this format as well `2011-12-03T10:15:30.9999`. Seconds and nano are optionals, so `2011-12-03T10:15` is a valid pattern user can wrote.
`LocalTime` uses [ISO_LOCAL_TIME](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#ISO_LOCAL_TIME) which is `10:15:30` format. It supports nanoseconds as well, so I added supports for this format as well `10:15:30.9999`. Seconds and nano are optionals, so `10:15` is a valid pattern user can wrote.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>
`

### CODE REVIEW

**Comment by yesamer** [comment]
jenkins retest this please

**Review by danielezonca — COMMENTED** [review]
Can you please update accordingly placeholder information too?
See
https://github.com/kiegroup/drools-wb/blob/17b04cc0361dc069fd126ff3723e2f89b2c4b39a/drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/utils/ScenarioSimulationUtils.java#L92-L111

**Comment by yesamer** [comment]
> Can you please update accordingly placeholder information too?
> See
> https://github.com/kiegroup/drools-wb/blob/17b04cc0361dc069fd126ff3723e2f89b2c4b39a/drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/utils/ScenarioSimulationUtils.java#L92-L111

@danielezonca Thank you for suggesting that. Done.

**Review by danielezonca — APPROVED** [review]
Thanks for your PR, approving :+1:

**Comment by yesamer** [comment]
jenkins do fdb

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/resources/org/drools/workbench/screens/scenariosimulation/client/resources/i18n/ScenarioSimulationEditorConstants.properties [hunk 1]
```diff
@@ -180,6 +180,8 @@ apply=Apply
 errorPopoverMessageFailedWithError=The expected value is ''{0}'' but the actual one is ''{1}''.
 errorPopoverMessageFailedWithException=Error message: {0}
 dateFormatPlaceholder=Insert date as yyyy-mm-dd
+dateTimeFormatPlaceholder=Insert datetime as yyyy-mm-ddThh:mm:ss
+timeFormatPlaceholder=Insert datetime as hh:mm:ss
```
Comment (jstastny-cz) [inline drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/resources/org/drools/workbench/screens/scenariosimulation/client/resources/i18n/ScenarioSimulationEditorConstants.properties:184]:
```suggestion
timeFormatPlaceholder=Insert time as hh:mm:ss
```

Comment (yesamer) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/resources/org/drools/workbench/screens/scenariosimulation/client/resources/i18n/ScenarioSimulationEditorConstants.properties]: Oopps! Thanks @jstastny-cz

**Comment by jstastny-cz** [comment]
**Testing feedback**
I've focused on all supported types in DataObject editor. From that
1. Date, OffsetDateTime - Unexcpected error in UI when trying to open -> not even possible to use the mvel expression definition
2. Long - long literals not considered valid inputs - e.g. 123L, 124L,...

IMHO both are bugs, 1. in a sense that at least an Unexpected error should not appear, but it could be easy win to implement similarly to LocalDate, wdyt?

**Comment by yesamer** [comment]
@jstastny-cz Point 1 could be easily added; Point 2: Which kind of error did you see?

**Comment by jstastny-cz** [comment]
jenkins do fdb

**Comment by jstastny-cz** [comment]
Seeing one issue with Date and OffsetDateTime now - when I don't specify input for Expect, the scenario succeeds, when I specify incorrect value, it fails correctly.
date

**Comment by yesamer** [comment]
@jstastny-cz I think it is a different issue - I suggest to create another ticket for that. Thank you.

**Review by jstastny-cz — APPROVED** [review]
Looks good, thanks.

**Comment by yesamer** [comment]
@kiegroup/gatekeepers Can you please merge this ensemble?

**Pull request merged**